blog/vscode-remote-development-setup-and-usage.mdx

```diff
@@ -0,0 +1,260 @@
+the necessary packages on your host machine (you will notice a notification
+stating “Setting up SSH Host on …”).
+
+And that’s it ! Hopefully once that is done, you'll already be set to develop on
+your host from your local machine. You’ll notice the icon in your Status bar is
+now showing the host you’re connected to, and you’ll be able to see the new
+connection in the Remote Explorer.
+
+### Password-less connection with VS Code Remote Development
+
+One thing that might get tiring using VS Code Remote, is that you might need to
+type your password each time you open a new VS Code window or each time your it
+gets disconnected from your host, if your internet or local network connection
+has hiccups for example. To solve this issue, I configured the SSH connection so
+that is uses a RSA key I created for this purpose. Here’s how it is done:
+
+#### Create a new key
+
+You can use `ssh-keygen` to create one. it is available on most Linux
+subsystems, and on Windows Powershell.
+
+```bash
+ssh-keygen -q -b 2048  -f "./<myhost>_rsa" -t rsa
+```
+
+This will create two files: `<myhost>_rsa` (the private key) and
+`<myhost>_rsa.pub` (the public key). The private key will stay on your client
+(typically in the `~/.ssh/` folder on Linux subsystems)
+
+The public key will need to be copied onto your host. On Linux systems you will
+need to add the content of the public key to the end of the
+`~/.ssh/authorized_keys` file. If the file doesn’t exist, just copy your public
+key in `~/.ssh/`, rename the file to `authorized_keys` and give it a permission
+of 600:
+
+```bash
+chmod 600 ~/.ssh/authorized_keys
+```
+
+#### Make VS Code use the key
+
+On your client IDE, go to the Remote Explorer, and click the cog icon next to
+the SSH section, and then open the config file shown in the popup by clicking on
+it:
+
+![SSH setting icon](./assets/vscode-remote-development-setup-and-usage/ssh-setting-icon.png)
+
+In this file, you’ll find the connection you just set up and its configuration.
+In order to make VSCode use the key you just created, add these lines to the
+configuration section:
+
+```bash
+Host <my_host>
+  HostName <my_host>
+  User <my_user>
+  Port 22
+  PreferredAuthentications publickey
+  IdentityFile "~/.ssh/<my_host>_rsa"
+```
+
+After this, VS Code should not ask for your user's password again !
+
+## Pros and cons
+
+Every method has its own advantages and disadvantages, and using this one
+certainly does not make any exception to this rule.
+
+### Pros
+
+As stated in during the installation part, one of the most convenient points of
+this method is the ease of use. Installation is really easy and well documented
+on VS Code docs, and within VS Code itself. Configuring it on a brand new
+machine should not take more than a few minutes, and you will immediately have
+access to all of your host’s installed extension without needing to authenticate
+to your Microsoft of Visual Studio account.
```

Comment:
**nitpick:** paragraphs makes it a bit harder to scan the pros/cons

**suggestion:** add a heading for each of them

```suggestion
#### Ease of use
As stated in during the installation part, one of the most convenient points of
this method is the ease of use. Installation is really easy and well documented
on VS Code docs, and within VS Code itself. Configuring it on a brand new
machine should not take more than a few minutes, and you will immediately have
access to all of your host’s installed extension without needing to authenticate
to your Microsoft of Visual Studio account.
```

etc. for others